src/common/server.ts

```diff
@@ -203,3 +228,29 @@ export async function restartServer(
 
   return newLSClient;
 }
+
+async function getRuffVersion(settings: ISettings): Promise<[number, number, number]> {
+  let command;
+  if (settings.path[0]) {
+    command = `${settings.path[0]} ${RUFF_VERSION_ARGS.join(" ")}`;
+  } else {
+    command = `${
+      settings.interpreter[0]
+    } ${EXPERIMENTAL_SERVER_SCRIPT_PATH} ${RUFF_VERSION_ARGS.join(" ")}`;
+  }
+  traceInfo(`Checking Ruff's version with command: ${command}`);
+  const execCommand = promisify(exec);
+  const { stdout } = await execCommand(command, {
+    cwd: settings.cwd,
+  });
+  const versionRegex = /ruff (\d+).(\d+).(\d+)/;
+  const matches = stdout.match(versionRegex) ?? [];
+  let major = Number(matches[1] ?? "X");
+  let minor = Number(matches[2] ?? "X");
+  let patch = Number(matches[3] ?? "X");
```

Comment:
I think we want to constrain this to integers only
```suggestion
  let major = parseInt(matches[1] ?? "X", 10);
  let minor = parseInt(matches[2] ?? "X", 10);
  let patch = parseInt(matches[3] ?? "X", 10);
```